internal/pkg/agent/application/dispatcher/dispatcher.go

```diff
@@ -21,6 +22,12 @@ import (
 
 type actionHandlers map[string]actions.Handler
 
+type persistedQueue interface {
```

Comment:
Does this need to be a private interface? Can we make this a real interface that is imported maybe from `queue`?

Reply:
I think this should remain defined in the dispatcher as these are the exported functions that it interacts with (it does not call `Cancel` or `Actions`).